### DISCUSSION TITLE
New rocket design

### DISCUSSION THREAD

**Discussion by SupSonAlv** [GuidodiPasquo/Actively-Stabilized-Model-Rocket-Simulator#11] [Q&A]
Getting errors when trying to draw a new body and the body does not draw.

I am getting some errors like below, why is this?

`\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 1948, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
   AeroVECTOR-master\AeroVECTOR-master\src\gui\gui_functions.py", line 1296, in slider_velocity
    self._draw_points()
    AeroVECTOR-master\AeroVECTOR-master\src\gui\gui_functions.py", line 957, in _draw_points
    self.canvas.delete(self.xcg_point_canvas, self.cp_point_canvas)
                       ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TabWithCanvas' object has no attribute 'xcg_point_canvas'`

**Comment by GuidodiPasquo** [comment]
Which steps did you take to make the rocket?, so I can reproduce the error

**Reply by SupSonAlv** [reply]
I got this error when I entered the values in the 'Parameter' section and switched to the 'Set rocket body' tab without creating a 'New File' file in the 'File' tab.
I actually figured out the cause of the problem.
I get this error if I switch to the 'Set rocket body' tab without saving a file with 'new file'. So that means. To avoid errors, a txt file should be created with 'New file' at the beginning to be precise.
-Or in order to prevent this usage, switching to other tabs should be prevented without creating a new file.

**Reply by GuidodiPasquo** [reply]
Yes, indeed, it's necessary to load or create a file first

**Comment by SupSonAlv** [comment]
So, where to start if you need to make an add-on that can prevent the user from switching to other tabs without creating and loading a file in the application?
